services-camel/src/test/java/edu/unc/lib/cdr/fulltext/FulltextRouterTest.java

```diff
@@ -0,0 +1,173 @@
+	private static final String IDENTIFIER = "org.fcrepo.jms.identifier";
+	private static final String RESOURCE_TYPE = "org.fcrepo.jms.resourceType";
+	private static final long timestamp = 1428360320168L;
+	private static final String userID = "bypassAdmin";
+	private static final String userAgent = "curl/7.37.1";
+	private static final String fileID = "/file1";
+	private final String eventTypes = EVENT_NS + "ResourceCreation";
+	private final String enhancementRoute = "CdrServiceFulltextExtraction";
+	private final String extractionRoute = "HasFulltext"; 
+	
+	@PropertyInject(value = "fcrepo.baseUri")
+	private static String baseUri;
+	
+	@EndpointInject(uri = "mock:fcrepo")
+	protected MockEndpoint resultEndpoint;
+	
+	@BeanInject(value = "repository")
+	private Repository repo;
+	
+	@Produce(uri = "direct:start")
+	protected ProducerTemplate template;
+	
+	@Before
+	public void init() {
+		PIDs.setRepository(repo);
+		when(repo.getBaseUri()).thenReturn(baseUri);
+	}
+	
+	@Override
+	protected AbstractApplicationContext createApplicationContext() {
+		return new ClassPathXmlApplicationContext("/service-context.xml", "/fulltext-context.xml");
+	}
+	
+	@Test
+	public void testRouteStart() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.filter").expectedMessageCount(1);
+		
+		createContext(enhancementRoute);
+		
+		template.sendBodyAndHeaders("", createEvent(fileID, eventTypes));
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testEventTypeFilter() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.filter").expectedMessageCount(0);
+		
+		createContext(enhancementRoute);
+		
+		Map<String, Object> headers = createEvent(fileID, eventTypes);
+		headers.put(EVENT_TYPE, "ResourceDeletion");
+		
+		template.sendBodyAndHeaders("", headers);
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testIdentifierFilter() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.filter").expectedMessageCount(0);
+		
+		createContext(enhancementRoute);
+
+		Map<String, Object> headers = createEvent(fileID, eventTypes);
+		headers.put(IDENTIFIER, "container");
+		
+		template.sendBodyAndHeaders("", headers);
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testResourceTypeFilter() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.filter").expectedMessageCount(0);
+		
+		createContext(enhancementRoute);
+
+		Map<String, Object> headers = createEvent(fileID, eventTypes);
+		headers.put(RESOURCE_TYPE, createResource( "http://bad.info/definitions/v9/repository#Fake" ).getURI());
+		
+		template.sendBodyAndHeaders("", headers);
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testFullTextExtractionFilterValidMimeType() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.extraction").expectedMessageCount(1);
+		createContext(extractionRoute);
+		
+		template.sendBodyAndHeaders("", createEvent(fileID, eventTypes));
```

Comment:
After you turn on the real mimetype filter you'll want to give it a real valid mimetype here